### ISSUE TITLE
"pip install robotframework-databaselibrary" fails if robot isn't already installed

### ISSUE THREAD

**Issue by edbrannin** [franz-see/Robotframework-Database-Library#42]
Reading DatabaseLibrary's `setup.py` fails if robotframework hasn't been installed yet, so it can't be installed from the same pip invocation that first installs robotframework.

It would be nice if:

1. This didn't cause the installation to fail
2. `setup.py` listed robotframework as a dependency

Steps to reproduce:

```
me@sandbox:~/tmp/robot-dblib-install-bug$ (
> echo robotframework
> echo robotframework-databaselibrary
> ) > requirements.txt
me@sandbox:~/tmp/robot-dblib-install-bug$ cat requirements.txt
robotframework
robotframework-databaselibrary
me@sandbox:~/tmp/robot-dblib-install-bug$ pip insta^C
me@sandbox:~/tmp/robot-dblib-install-bug$ virtualenv env
New python executable in env/bin/python
Installing setuptools, pip...done.
me@sandbox:~/tmp/robot-dblib-install-bug$ . env/bin/activate
(env)me@sandbox:~/tmp/robot-dblib-install-bug$ pip install -r requirements.txt
```

Error output:
```
Downloading/unpacking robotframework (from -r requirements.txt (line 1))
  Downloading robotframework-2.8.6.tar.gz (378kB): 378kB downloaded
  Running setup.py (path:/home/me/tmp/robot-dblib-install-bug/env/build/robotframework/setup.py) egg_info for package robotframework

    no previously-included directories found matching 'src/robot/htmldata/testdata'
Downloading/unpacking robotframework-databaselibrary (from -r requirements.txt (line 2))
  Downloading robotframework-databaselibrary-0.6.tar.gz
  Running setup.py (path:/home/me/tmp/robot-dblib-install-bug/env/build/robotframework-databaselibrary/setup.py) egg_info for package robotframework-databaselibrary
    Traceback (most recent call last):
      File "<string>", line 17, in <module>
      File "/home/me/tmp/robot-dblib-install-bug/env/build/robotframework-databaselibrary/setup.py", line 25, in <module>
        from DatabaseLibrary import __version__
      File "/home/me/tmp/robot-dblib-install-bug/env/build/robotframework-databaselibrary/src/DatabaseLibrary/__init__.py", line 15, in <module>
        from connection_manager import ConnectionManager
      File "/home/me/tmp/robot-dblib-install-bug/env/build/robotframework-databaselibrary/src/DatabaseLibrary/connection_manager.py", line 16, in <module>
        from robot.api import logger
    ImportError: No module named robot.api
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):

  File "<string>", line 17, in <module>

  File "/home/me/tmp/robot-dblib-install-bug/env/build/robotframework-databaselibrary/setup.py", line 25, in <module>

    from DatabaseLibrary import __version__

  File "/home/me/tmp/robot-dblib-install-bug/env/build/robotframework-databaselibrary/src/DatabaseLibrary/__init__.py", line 15, in <module>

    from connection_manager import ConnectionManager

  File "/home/me/tmp/robot-dblib-install-bug/env/build/robotframework-databaselibrary/src/DatabaseLibrary/connection_manager.py", line 16, in <module>

    from robot.api import logger

ImportError: No module named robot.api

----------------------------------------
Cleaning up...
Command python setup.py egg_info failed with error code 1 in /home/me/tmp/robot-dblib-install-bug/env/build/robotframework-databaselibrary
Storing debug log for failure in /home/me/.pip/pip.log
(env)me@sandbox:~/tmp/robot-dblib-install-bug$

```

**Comment by edbrannin** [comment]
Related: see [this Stack Overflow answer](http://stackoverflow.com/a/2073599/25625).

**Comment by jerry57** [comment]
Can you please submit a pull request with the changes needed?  This does sound like it will resolve the issue nicely.

**Comment by edbrannin** [comment]
I'm finishing that up now -- shouldn't this maybe also declare a dependency on robotframework?

**Comment by edbrannin** [comment]
Oh, I see you had already tried to depend on robotframework, but [it's not working](http://stackoverflow.com/a/15835083/25625).

**Comment by edbrannin** [comment]
(I've also made a branch for the dependency fix, but it depends on PR #44's changes so I'm going to let that get sorted first.)

**Comment by jerry57** [comment]
Closing, this issue should now be resolved and will get PyPi  updated soon.
